src/pages/workspace/WorkspaceInitialPage.tsx

```diff
@@ -167,7 +186,24 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, reimbursementAcc
 
     const protectedCollectPolicyMenuItems: WorkspaceMenuItem[] = [];
 
-    if (policy?.areDistanceRatesEnabled) {
+    // We only update feature states if they aren't pending.
+    // These changes are made to synchronously change feature states along with FeatureEnabledAccessOrNotFoundComponent.
+    useEffect(() => {
+        setFeatureStates((currentFeatureStates) => {
+            const newFeatureStates = {} as PolicyFeatureStates;
+            (Object.keys(policy?.pendingFields ?? {}) as PolicyFeatureName[]).forEach((key) => {
```

Comment:
I think we should always avoid casting (`as`) since it breaks type safety. Was it really necessary?